### PR TITLE
Pull Request for Issue1956: IDEAS update

### PR DESCRIPTION
**Pull request by dretrex**
This is a simple pull request where I have brought IDEAS up to date with a couple small fixes.

### CODE REVIEW

File: source/acquaman/IDEAS/IDEASXASScanActionController.cpp [hunk 1]
```diff
@@ -177,12 +177,11 @@ void IDEASXASScanActionController::buildScanControllerImplementation()
 			scan_->addAnalyzedDataSource(newRegion, false, true);
 			detector->addRegionOfInterest(region);
 
-			QString regionName = regionAB->name().replace(" ","_");
-			AM1DCalibrationAB* normalizedRegion = new AM1DCalibrationAB(QString("Norm%1").arg(regionName));
-			normalizedRegion->setDescription(QString("Norm%1").arg(regionName));
+			AM1DCalibrationAB* normalizedRegion = new AM1DCalibrationAB(QString("norm_%1").arg(newRegion->name()));
```
Comment (XuetaoLiu-CLS) [inline source/acquaman/IDEAS/IDEASXASScanActionController.cpp]: newRegion->name() is the regionAB->name() without space (`regionAB->name().remove(' ')`). 
the removed `regionName` is the regionAB->name() with `_` replacing space (`regionAB->name().replace(" ", "_")`). 

Therefore, the changes is slightly different with the original ones (name without `_`). Is this what we want?

File: source/ui/IDEAS/IDEASPersistentView.cpp [hunk 2]
```diff
@@ -88,132 +88,125 @@ IDEASPersistentView::IDEASPersistentView(QWidget *parent) :
...
-				jjSlitLayout_->addWidget(jjSlitsRight_,1,2,1,1);
-
-
-
-
-
-			jjSlitGroupBox_->setLayout(jjSlitLayout_);
-
-
+	QGroupBox *jjSlitGroupBox = new QGroupBox;
```
Comment (XuetaoLiu-CLS) [inline source/ui/IDEAS/IDEASPersistentView.cpp]: we have a `CLSJJSlitsView`. Can't it meet the slit display/operation requirements with IDEAS?

**Comment by XuetaoLiu-CLS** [comment]
Looks it did what it said. It might be able to reuse some existing class (`CLSJJSlitView`)

Time Estimate: 5 minutes

File: source/acquaman/IDEAS/IDEASXASScanActionController.cpp (hunk 1 above)
Comment (dretrex) [inline source/acquaman/IDEAS/IDEASXASScanActionController.cpp]: Yes, without that the standard code that @epengr was using can't find the region data source.

File: source/ui/IDEAS/IDEASPersistentView.cpp (hunk 2 above)
Comment (dretrex) [inline source/ui/IDEAS/IDEASPersistentView.cpp]: Hilariously, no.  The JJSlits for IDEAS actually work differently under the hood.  You directly tell it to move the gap and center whereas other JJ slit implementations let you control the actual blades.

**Pull request merged**